crates/core/tedge_api/src/mqtt_topics.rs

```diff
@@ -334,6 +335,32 @@ impl FromStr for EntityTopicId {
     }
 }
 
+impl<'de> Deserialize<'de> for EntityTopicId {
+    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
+    where
+        D: Deserializer<'de>,
+    {
+        struct EntityTopicIdVisitor;
+
+        impl Visitor<'_> for EntityTopicIdVisitor {
+            type Value = EntityTopicId;
+
+            fn expecting(&self, formatter: &mut std::fmt::Formatter) -> std::fmt::Result {
+                formatter.write_str("four segments entity topic id")
+            }
+
+            fn visit_str<E>(self, value: &str) -> Result<EntityTopicId, E>
+            where
+                E: serde::de::Error,
+            {
+                EntityTopicId::from_str(value).map_err(serde::de::Error::custom)
+            }
+        }
+
+        deserializer.deserialize_str(EntityTopicIdVisitor)
```

Comment:
How about a simpler impl?
```suggestion
        let s = String::deserialize(deserializer)?;
        EntityTopicId::from_str(&s).map_err(serde::de::Error::custom)
```

Reply:
Yeah, you're right. Applied in [08583c3](https://github.com/thin-edge/thin-edge.io/pull/3578/commits/08583c3180c93a1573157c0da46e6c432df18782)